go.mod

```diff
@@ -42,20 +43,22 @@ require (
 	github.com/pborman/uuid v1.2.0
 	github.com/performancecopilot/speed v3.0.0+incompatible
 	github.com/pkg/errors v0.8.1
-	github.com/prometheus/client_golang v1.3.0
+	github.com/prometheus/client_golang v1.5.1
 	github.com/samuel/go-zookeeper v0.0.0-20190923202752-2cc03de413da
 	github.com/sirupsen/logrus v1.4.2
 	github.com/smartystreets/goconvey v1.6.4 // indirect
 	github.com/sony/gobreaker v0.4.1
 	github.com/streadway/amqp v0.0.0-20190827072141-edfb9018d271
 	github.com/streadway/handy v0.0.0-20190108123426-d5acb3125c2a
-	go.etcd.io/etcd v0.0.0-20191023171146-3cf2f69b5738
+	go.etcd.io/etcd/client/pkg/v3 v3.5.0-beta.4
+	go.etcd.io/etcd/client/v2 v2.305.0-beta.4
+	go.etcd.io/etcd/client/v3 v3.5.0-beta.4
```

Comment:
I can't say I understand why all of these are here. Maybe a quick summary?

Reply:
`go.etcd.io/etcd/client/v2`: used by `sd/etcd` (v2 client became a separate module)
`go.etcd.io/etcd/client/v3`: used by `sd/etcdv3`
`go.etcd.io/etcd/client/pkg/v3`: some common client code has been extracted to separate a pkg module (go-kit uses the transport package from there)

etcd decided to slice up the project into smaller modules.

Does that help?